### PR TITLE
feat: sub commands

### PR DESCRIPTION
**Pull request by DjDeveloperr**

### CODE REVIEW

File: src/rest/endpoints.ts [hunk 1]
```diff
@@ -340,8 +341,27 @@ export class RESTEndpoints {
...
+  async getGuildAuditLog(
+    guildId: string,
+    params: {
+      userId?: string
+      actionType?: AuditLogEvents
+      before?: string
+      limit?: number
+    }
+  ): Promise<AuditLogPayload> {
+    let q = ''
```
Comment (AkiaCode) [inline src/rest/endpoints.ts:353]: I think it would be better to use searchParams than q

File: src/structures/guild.ts [hunk 2]
```diff
@@ -49,6 +53,7 @@ import { DiscordAPIError } from '../rest/mod.ts'
 import type { ImageFormats, ImageSize } from '../types/cdn.ts'
 import { ImageURL } from './cdn.ts'
 import type { GuildSlashCommandsManager } from '../interactions/slashCommand.ts'
+import { toCamelCase } from '../utils/snake_case.ts'
```
Comment (AkiaCode) [inline src/structures/guild.ts:56]: please make it as simple as possible and in camelCase.

File: src/commands/command.ts [hunk 3]
```diff
@@ -492,15 +558,25 @@ export class CommandsManager {
...
+    if (cmd.name === '' && CmdClass !== undefined) {
+      let name = CmdClass.name
+      if (
+        name.toLowerCase().endsWith('command') &&
+        name.toLowerCase() !== 'command'
+      )
+        name = name.substr(0, name.length - 'command'.length).trim()
+      cmd.name = name
+    }
     if (cmd.name === '') throw new Error('Command has no name')
```
Comment (AkiaCode) [inline src/commands/command.ts:580]: I think `!cmd.length` looks better than `cmd.name === ''`

Comment (DjDeveloperr) [reply src/commands/command.ts:580]: Linter will compain on `!cmd.length` and say to use `cmd.length === 0` 🤔

Comment (AkiaCode) [reply src/commands/command.ts:580]: Oops, Okay

File: src/rest/endpoints.ts (hunk 1 above)
Comment (Helloyunho) [reply src/rest/endpoints.ts:353]: I think query(or q) makes more sence

File: src/utils/snake_case.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  const parts = name.split('_')
+  return parts
+    .map((e, i) =>
+      i === 0 ? e : e === 'id' ? 'ID' : `${e[0].toUpperCase()}${e.slice(1)}`
+    )
+    .join('')
+}
+
+export function toCamelCase(data: any): any {
+  if (typeof data === 'object' && Array.isArray(data))
```
Comment (Helloyunho) [inline src/utils/snake_case.ts:11]: You could use `Array.isArray()` only

**Review by Helloyunho — CHANGES_REQUESTED** [review]

**Review by Helloyunho — APPROVED** [review]
LGTM!

**Pull request merged**